src/alchemlyb/parsing/namd.py

```diff
@@ -25,72 +108,162 @@ def extract_u_nk(fep_file, T):
+                            lambda_idws_at_start = all_lambdas[l2_idx + 1]
+
+                        logger.warning(f'Warning: {fep_file} has IDWS data but lambda_idws not specified.')
+                        logger.warning(f'         lambda1 = {lambda1}, lambda2 = {lambda2}; inferring lambda_idws to be {lambda_idws_at_start}')
+
+                    if lambda_idws_at_start is not None:
+                        # Mimic classic DWS data
+                        # Arbitrarily match up fwd and bwd comparison energies on the same times
+                        # truncate extra samples from whichever array is longer
+                        win_de_back_arr = beta * np.asarray(win_de_back)
+                        n = min(len(win_de_back_arr), len(win_de_arr))
+
+                        tempDF = pd.DataFrame({
+                            'time': win_ts_arr[:n],
+                            'fep-lambda': np.full(n,lambda1),
+                            lambda1: 0,
+                            lambda2: win_de_arr[:n],
+                            lambda_idws_at_start: win_de_back_arr[:n]})
+                        # print(f"{fep_file}: IDWS window {lambda1} {lambda2} {lambda_idws_at_start}")
+                    else:
+                        # print(f"{fep_file}: Forward-only window {lambda1} {lambda2}")
+                        # create dataframe of times and work values
+                        # this window's data goes in row LAMBDA1 and column LAMBDA2
+                        tempDF = pd.DataFrame({
+                            'time': win_ts_arr,
+                            'fep-lambda': np.full(len(win_de_arr), lambda1),
+                            lambda1: 0,
+                            lambda2: win_de_arr})
+
+                    # join the new window's df to existing df
+                    u_nk = pd.concat([u_nk, tempDF], sort=True)
+
+                    # reset values for next window of fepout file
+                    win_de = []
+                    win_ts = []
+                    win_de_back = []
+                    win_ts_back = []
+                    parsing = False
+                    has_idws = False
+                    lambda1_at_start, lambda2_at_start, lambda_idws_at_start = None, None, None
+
+                # append work value from 'dE' column of fepout file
+                if parsing:
+                    if l[0] == 'FepEnergy:':
+                        win_de.append(float(l[6]))
+                        win_ts.append(float(l[1]))
+                    elif l[0] == 'FepE_back:':
+                        win_de_back.append(float(l[6]))
+                        win_ts_back.append(float(l[1]))
+
+                # Turn parsing on after line 'STARTING COLLECTION OF ENSEMBLE AVERAGE'
+                if '#STARTING' in l:
+                    parsing = True
+
+    if len(win_de) != 0 or len(win_de_back) != 0:
+        logger.warning('Trailing data without footer line (\"#Free energy...\"). Interrupted run?')
```

Comment:
Is this harmless or should it error?

If it's harmless then add a `# pragma: nocover` to this block.

Reply:
Should be harmless, this parser is meant to handle interrupted runs, but the user should know what's going on.

Reply:
Ok, then mark it as not covered by tests. I *think* you can just modify the line
```python
    if len(win_de) != 0 or len(win_de_back) != 0:   # pragma: no cover
```
and it should exclude the whole block. But see https://coverage.readthedocs.io/en/latest/excluding.html for details.